### ISSUE TITLE
sync_streaming_ping_pong_test failed on macos

### ISSUE THREAD

**Issue by yang-g** [grpc/grpc#3910]
https://grpc-testing.appspot.com/job/gRPC_master/12020/config=dbg,language=c++,platform=macos/console

E1021 13:30:42.182922000 140735213904640 driver.cc:226] assertion failed: server->stream->Read(&server_status)

*******************************
Caught signal 6
  no backtrace

FAILED: bins/dbg/qps_test [ret=-6, pid=67832]
FAILED: Some tests failed
+ true
+ '[' '!' -e reports/index.html ']'
+ mkdir -p reports
+ echo 'No reports generated.'
[htmlpublisher] Archiving HTML reports...
[htmlpublisher] Archiving at BUILD level /jenkins/workspace/gRPC_master/config/dbg/language/c++/platform/macos/reports to /home/jenkins/jenkins_home/jobs/gRPC_master/configurations/axis-config/dbg/axis-language/c++/axis-platform/macos/builds/12020/htmlreports/HTML_Report
[copy-to-slave] Copying 'report.xml', excluding nothing, from 'file:/jenkins/workspace/gRPC_master/config/dbg/language/c++/platform/macos/' on 'hudson.slaves.DumbSlave[grpc-mac-node2]' to 'file:/home/jenkins/jenkins_home/jobs/gRPC_master/workspace/config/dbg/language/c++/platform/macos/' on the master.
Recording test results
Build step 'Publish JUnit test result report' changed build result to UNSTABLE
Finished: UNSTABLE

**Comment by yang-g** [comment]
qps_test has the same problem:

https://grpc-testing.appspot.com/job/gRPC_master/12020/config=dbg,language=c++,platform=macos/consoleText

**Comment by a11r** [comment]
another failure:

https://grpc-testing.appspot.com/job/gRPC_master/12375/config=dbg,language=c++,platform=macos/testReport/(root)/tests/bins_dbg_sync_streaming_ping_pong_test/

**Comment by dgquintas** [comment]
Still failing:

```
I1102 12:48:58.794363000 140735213904640 sync_streaming_ping_pong_test.cc:51] Running Synchronous Streaming Ping Pong
D1102 12:48:58.794996000 140735213904640 test_config.c:155] test slowdown: machine=1.000000 build=1.000000 total=1.000000
E1102 12:48:58.796632000 140735213904640 tcp_client_posix.c:171] failed to connect to 'ipv6:[::1]:32767': socket error: connection refused
E1102 12:48:58.798218000 140735213904640 tcp_client_posix.c:171] failed to connect to 'ipv6:[::1]:32767': socket error: connection refused
E1102 12:48:58.800725000 140735213904640 tcp_client_posix.c:171] failed to connect to 'ipv6:[::1]:32767': socket error: connection refused
E1102 12:48:58.801732000 140735213904640 tcp_client_posix.c:171] failed to connect to 'ipv6:[::1]:32767': socket error: connection refused
D1102 12:48:58.804657000 4324679680 grpc_profiler.c:48] You do not have google-perftools installed, profiling is disabled [for qps_server.prof]
D1102 12:48:58.804682000 4324679680 grpc_profiler.c:52] To install on ubuntu: sudo apt-get install google-perftools libgoogle-perftools-dev
I1102 12:48:58.807648000 140735213904640 driver.cc:206] Warming up
I1102 12:49:03.808309000 140735213904640 driver.cc:212] Starting
E1102 12:49:33.811954000 140735213904640 driver.cc:229] assertion failed: client->stream->Read(&client_status)
```

**Comment by vjpai** [comment]
I believe this flake has been fixed by #4018. Will watch and see.

**Comment by vjpai** [comment]
I declare this closed based on a lack of failures in recent master builds since #4018 was merged.
